nexus/lib/qmcpack_input.py

```diff
@@ -4811,15 +4813,121 @@ def generate_determinantset_old(type           = 'bspline',
             #Should be modified
             QmcpackInput.class_error('excitation must be a tuple or list with with two elements\nthe first element must be either "up" or "down"\nand the second element must be integers separated by spaces, e.g. "-216 +217"\nyou provided: {0}'.format(excitation))
         #end if
-        
+
         spin_channel,excitation = excitation
-        
         if spin_channel=='up':
-            det = dset.get('updet')
+            sdet = dset.get('updet')
         elif spin_channel=='down':
-            det = dset.get('downdet')
+            sdet = dset.get('downdet')
+        elif spin_channel=='singlet' or spin_channel=='triplet':
+
+            # Is multi-det WF appropriate?
+            if elns.down_electron.count != elns.up_electron.count:
+                QmcpackInput.class_error('The \'singlet\' and \'triplet\' excitation types assume number of up and down electrons is the same. Otherwise, one should use \'up\' or \'down\' types.\nFor your system: Nup={} and Ndown={}'.format(elns.up_electron.count,elns.down_electron.count))
```

Comment:
Doesn't this requirement depend on the reference state (e.g. if it is already singlet-like or triplet-like)?

Reply:
Presently, the code assumes that the ground state is a closed shell singlet and the user wants to create a pure singlet or pure triplet excited state

Reply:
OK for now, can expand in the future.  Change the message to "types currently assume".

Reply:
Resolved.